src/extension.ts

```diff
@@ -203,13 +204,22 @@ async function removeSession(sessionId: string): Promise<void> {
 
 export async function activate(context: extensionApi.ExtensionContext): Promise<void> {
   console.log('starting redhat-authentication extension');
+  const statusBarItem = new SSOStatusBarItem();
+  statusBarItem.show()
+  context.subscriptions.push(statusBarItem);
   if (!authenticationServicePromise) {
     authenticationServicePromise = RedHatAuthenticationService.build(context, getAuthConfig())
       .then(service => {
         context.subscriptions.push(service);
-        service.initialize();
-        return service;
-    });
+        return service.initialize().then(() => service);
```

Comment:
I think here we reached from readability too many then then then

Reply:
I'll move it out to async function.